libs/rpc/connection.js

```diff
@@ -17,11 +24,27 @@ var Connection = function (driver) {
  * @return {Promise}
  */
 Connection.prototype.callAsync = function (method) {
-	var request = {
-		method : method,
-		data : Array.prototype.slice(arguments, 1),
-	};
-	return this._driver.send(request);
+	// Create request
+	var request = Request.createRequest(
+		method,
+		Array.prototype.slice(arguments, 1)
+	);
+
+	// Make promise for request
+	var promise = {};
```

Comment:
Rename this variable to something different then promise, since constructions like promise.promise are not easy to read / understand. You could use promiseData or something similar.